### PR TITLE
refactor(chaoslibs): Refactored the generic pod experiments, cassandra and kafka experiments

### PR DESCRIPTION
**Pull request by ispeakc0de**
Signed-off-by: shubhamchaudhary <shubham@chaosnative.com>

Refactor all the generic pod experiments, Cassandra and Kafka experiments for the following:
- Removed all the redundant code and convert them to reusable functions
- Follow go-lint standards
- update chaos iteration logic

Tested for:
- container-kill :heavy_check_mark: 
- disk-fill :heavy_check_mark: 
- kafka-pod-delete :heavy_check_mark: 
- network-chaos :heavy_check_mark: 
- pod-autoscaler :heavy_check_mark: 
- pod-cpu-hog :heavy_check_mark: 
- pod-delete :heavy_check_mark: 
- dns-chaos :heavy_check_mark: 
- pod-memory-hog :heavy_check_mark: 
- cassandra-pod-delete :heavy_check_mark:

### CODE REVIEW

File: pkg/utils/common/pid.go [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+		PID, err = parsePIDFromJSON(out, runtime)
+		if err != nil {
+			log.Errorf(fmt.Sprintf("[cri]: Failed to parse json from crictl output: %s", string(out)))
+			return 0, err
+		}
+	default:
+		return 0, errors.Errorf("%v container runtime not suported", runtime)
+	}
+
+	log.Info(fmt.Sprintf("[cri]: Container ID=%s has process PID=%d", containerID, PID))
```
Comment (uditgaurav) [inline pkg/utils/common/pid.go:90]:
```suggestion
	log.Info(fmt.Sprintf("[Info]: Container ID=%s has process PID=%d", containerID, PID))
```

File: pkg/utils/common/pid.go [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+				return 0, err
+			}
+			pid = resp.Info.PID
+		}
+	default:
+		return 0, errors.Errorf("[cri]: No supported container runtime, runtime: %v", runtime)
+	}
+	if pid == 0 {
+		return 0, errors.Errorf("[cri]: No running target container found, pid: %d", pid)
+	}
```
Comment (uditgaurav) [inline pkg/utils/common/pid.go:137]:
```suggestion
		return 0, errors.Errorf("[Error]: Container runtime not supported, runtime: %v", runtime)
	}
	if pid == 0 {
		return 0, errors.Errorf("[Error]: No running target container found, pid: %d", pid)
	}
```

File: pkg/utils/common/pods.go [hunk 3]
```diff
@@ -272,9 +265,71 @@ func DeleteAllHelperPodBasedOnJobCleanupPolicy(podLabel string, chaosDetails *ty
...
+
+	log.Infof("container ID of %v container, containerID: %v", targetContainer, containerID)
+	return containerID, nil
+}
+
+// CheckContainerStatus checks the status of the application container
+func CheckContainerStatus(appNamespace, appName string, clients clients.ClientSets) error {
+	return retry.
+		Times(90).
+		Wait(2 * time.Second).
```
Comment (uditgaurav) [inline pkg/utils/common/pods.go:320]: Hardcoded Values for delay and retries?

File: pkg/utils/common/pods.go [hunk 4]
```diff
@@ -272,9 +265,71 @@ func DeleteAllHelperPodBasedOnJobCleanupPolicy(podLabel string, chaosDetails *ty
...
+	pod, err := clients.KubeClient.CoreV1().Pods(chaosNamespace).Get(chaosPodName, v1.GetOptions{})
+	if err != nil {
+		return "", err
+	}
+	return pod.Spec.ServiceAccountName, nil
+}
+
+//GetTargetContainer will fetch the container name from application pod
+//This container will be used as target container
+func GetTargetContainer(appNamespace, appName string, clients clients.ClientSets) (string, error) {
```
Comment (uditgaurav) [inline pkg/utils/common/pods.go:285]: How about `GetTargetContainerAsFirstContainer`?

**Comment by uditgaurav** [comment]
Some Minor comments please check the conflicts.

**Comment by uditgaurav** [comment]
Do we also need these changes in SDK?

**Review by uditgaurav — DISMISSED** [review]

**Review by ksatchit — APPROVED** [review]

**Review by ksatchit — APPROVED** [review]

**Review by uditgaurav — APPROVED** [review]

**Pull request merged**